### DISCUSSION TITLE
Custom UI component

### DISCUSSION THREAD

**Discussion by fulviodan** [writer/writer-framework#62] [Q&A]
Hello, first of all I want to say that it is a wonderful project:-) 

I was wondering if there is a way of writing custom UI components with their specific logic. Let's say, for example, I want to create an image annotation component that will be shown on the "toolkit" sidebar on the left or to extend the dataframe component. Is there a way to provide, in the backend, a vue component or something like that? Thank you in advance for your help!

**Comment by ramedina86** [comment]
Hi, I'm glad you like Streamsync!

Custom components are currently under development, they'll be available for the next release.

Right now the only way to add components would be to modify the frontend (in `/ui`) and recompile it, which is cumbersome.

**Reply by fulviodan** [reply]
thanks!
